.github/workflows/codeql-analysis.yml

```diff
@@ -32,38 +24,43 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        language: ['javascript', 'python']
-        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python' ]
-        # Learn more:
-        # https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#changing-the-languages-that-are-analyzed
+        include:
+          - language: 'go'
+            runs-on: ['self-hosted', '1ES.Pool=ubuntu2204']
+          - language: 'javascript'
+            runs-on: ['ubuntu-latest']
+          - language: 'python'
+            runs-on: ['ubuntu-latest']
 
     steps:
+    - name: Install dependencies
+      if: matrix.language == 'go'
+      run: |
+        sudo apt-get update
+        sudo apt-get install libbtrfs-dev libgpgme-dev libdevmapper-dev
+
     - name: Checkout repository
       uses: actions/checkout@v3
 
-    # Initializes the CodeQL tools for scanning.
+    - name: Set up Go
+      if: matrix.language == 'go'
+      uses: actions/setup-go@v4
+      with:
+        go-version: 1.18
+
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v2
       with:
         languages: ${{ matrix.language }}
-        # If you wish to specify custom queries, you can do so here or in a config file.
-        # By default, queries listed here will override any specified in a config file.
-        # Prefix the list here with "+" to use these queries and those in the config file.
-        # queries: ./path/to/local/query, your-org/your-repo/queries@main
 
-    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
-    # If this step fails, then you should remove it and run the build manually (see below)
     - name: Autobuild
+      if: matrix.language != 'go'
       uses: github/codeql-action/autobuild@v2
 
-    # ℹ️ Command-line programs to run using the OS shell.
-    # 📚 https://git.io/JvXDl
-
-    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
-    #    and modify them (or add more) to build your code if your project
-    #    uses a compiled language
+    - name: Build Go
+      if: matrix.language == 'go'
+      run: |
+        go build -v ./cmd/aro
```

Comment:
now that we have runners, do we want to let autobuild do everything for us?

Reply:
Will autobuild work for this? I haven't tested it, but can.

Reply:
yeah that's what we previously used.  

https://github.com/mbarnes/ARO-RP/blob/4094ce29905a13c399413cb48cc98a9ee4440f74/.github/workflows/codeql-analysis.yml#L35-L57

Reply:
Removed the specific build step. All I'm doing now that's go specific is ensuring we have the right go version, and installing the Linux headers we need